# 2. Observation of the Datasets

### 0. List of Contents
#### 1. Importing Libraries and Dataframes
#### 2.  Exploring DF
#### 3. Data Cleaning

In [1]:
#import libraries
import pandas as pd
import numpy as np
import os

In [2]:
# Creating path
path=r'/Users/fru.s/Documents/CF - Data Analytics/2.6. Sourcing Open Data/Open Source Data Analysis/02 Data/Prepared Data/'

In [3]:
# Importing MARITAL_STATUS_BY_AGE as ms
ms = pd.read_pickle(os.path.join(path,'ms_clean.pkl'))

In [4]:
# Importing EVER_MARRIED as em
em = pd.read_pickle(os.path.join(path,'em_clean.pkl'))

In [16]:
# Importing SMAM as em
smam = pd.read_pickle(os.path.join(path,'smam_clean.pkl'))

In [6]:
# Importing CURRENTLY MARRIED as cm
cm = pd.read_pickle(os.path.join(path,'cm_clean.pkl'))

### 2. Exploring DFs

In [4]:
#Displaying max columns with pd.set_option
pd.set_option('display.max_columns',None)

In [7]:
ms.describe()

YearStart        YearEnd      DataValue
count  256334.000000  256334.000000  256334.000000
mean     1997.402596    1997.497148      21.852470
std        14.022351      14.060471      30.025002
min      1960.000000    1960.000000       0.000000
25%      1986.000000    1986.000000       1.100000
50%      2000.000000    2000.000000       6.010000
75%      2010.000000    2010.000000      32.430000
max      2019.000000    2019.000000     100.000000

#### The DataValue of the ms dataset doesn't tell much by itself, without the marital status column

In [9]:
ms.head(10)

Country or area  YearStart  YearEnd  Sex MaritalStatus AgeGroup  DataValue  \
0     Afghanistan       1972     1974  Men      Divorced  [15-19]        0.0   
1     Afghanistan       1972     1974  Men      Divorced  [20-24]        0.0   
2     Afghanistan       1972     1974  Men      Divorced  [25-29]        0.0   
3     Afghanistan       1972     1974  Men      Divorced  [30-34]        0.2   
4     Afghanistan       1972     1974  Men      Divorced  [35-39]        0.1   
5     Afghanistan       1972     1974  Men      Divorced  [40-44]        0.1   
6     Afghanistan       1972     1974  Men      Divorced  [45-49]        0.0   
7     Afghanistan       1972     1974  Men      Divorced  [50-54]        0.0   
8     Afghanistan       1972     1974  Men      Divorced  [55-59]        0.0   
9     Afghanistan       1972     1974  Men      Divorced  [60-64]        0.0   

           Data Source Region Name Sub-region Name  
0  National statistics        Asia   Southern Asia  
1  National statistics        Asia   Southern Asia  
2  National statistics        Asia   Southern Asia  
3  National statistics        Asia   Southern Asia  
4  National statistics        Asia   Southern Asia  
5  National statistics        Asia   Southern Asia  
6  National statistics        Asia   Southern Asia  
7  National statistics        Asia   Southern Asia  
8  National statistics        Asia   Southern Asia  
9  National statistics        Asia   Southern Asia

In [10]:
ms.groupby('MaritalStatus').agg({'DataValue': ['mean', 'min', 'max']})

DataValue               
                                                    mean    min     max
MaritalStatus                                                          
Consensual union                                9.954759   0.00   90.38
Consensual union, not living together           8.384330   0.00   62.75
Currently not married                          31.491429  12.50   81.53
Currently not married nor in consensual union  30.745556   2.70   96.50
Divorced                                        4.209827   0.00   87.60
Divorced or Separated                           3.124538   0.00   71.01
Divorced or Separated or Widowed                6.744706   0.10   61.10
Divorced or Widowed                            10.316774   0.22   60.22
Ever married                                   69.894393   0.00   99.70
Living together                                13.091764   0.00   81.80
Marriage contract                               2.890500   0.01   36.78
Married                                        54.734348   0.00  100.00
Married monogamous                             56.087200   1.10   81.80
Married or Living together                     56.582545   0.00   98.53
Married or in consensual union                 61.413127   0.00   98.23
Married or married but separated               53.824677   0.00   89.19
Married polygamous                              5.972857   0.10   19.04
Married, in consensual unions or separated     57.717245   0.00   85.70
Never married                                  21.326024   0.00  100.00
Not in union                                    1.807390   0.00   20.13
Not living together                             3.705629   0.00   28.10
Registred partnership                           0.824743   0.00    9.71
Separated                                       2.336974   0.00   71.24
Separated from consensual union                 4.014444   0.02    8.00
Separated from marriage                         0.718519   0.00    1.38
Single                                         30.393761   0.00  100.00
Single or in consensual unions                 27.595714   4.38  100.00
Visiting partner                                5.690096   0.16   49.30
Widowed                                         8.643314   0.00   90.90
Widowed or divorced                            10.782161   0.00   81.55
Widowed or separated                           27.475000   0.30   78.00
Widowed, divorced or separated                 11.933830   0.00   73.36

In [13]:
em.head()

Country or area  YearStart  YearEnd  Sex AgeGroup  DataValue  \
0     Afghanistan       1972     1974  Men  [15-19]        7.7   
1     Afghanistan       1972     1974  Men  [20-24]       32.6   
2     Afghanistan       1972     1974  Men  [25-29]       61.4   
3     Afghanistan       1972     1974  Men  [30-34]       83.0   
4     Afghanistan       1972     1974  Men  [35-39]       91.2   

           Data Source Region Name Sub-region Name  
0  National statistics        Asia   Southern Asia  
1  National statistics        Asia   Southern Asia  
2  National statistics        Asia   Southern Asia  
3  National statistics        Asia   Southern Asia  
4  National statistics        Asia   Southern Asia

In [14]:
#I need to aggreagte for EM as well, because the DataValue by itself doesn't mean much
em.groupby('AgeGroup').agg({'DataValue': ['mean', 'min', 'max']})

DataValue               
               mean    min     max
AgeGroup                          
[0-14]    24.845000   0.01   84.91
[10-14]    0.433250   0.00  100.00
[12-14]    3.840000   2.37    5.26
[15-19]    7.487093   0.00  100.00
[16-19]    0.315000   0.18    0.45
[20-24]   31.872404   0.00  100.00
[25-29]   59.868556   3.27  100.00
[30-34]   77.169064   2.26  100.00
[35-39]   85.098849  30.94  100.20
[40-44]   88.906106  40.57  100.47
[45+]     98.350000  98.19   98.51
[45-49]   90.953343  44.12  100.44
[50+]     96.222500  81.40  100.00
[50-54]   91.477566  41.63  100.33
[55+]     94.550556  59.03   99.35
[55-59]   92.362404  41.26  102.80
[60+]     95.427101  80.83   99.72
[60-64]   92.645790  26.18  101.23
[65+]     90.829252  41.77  101.64
[65-69]   93.274009  34.81  100.83
[70+]     95.450000  42.34  100.00
[70-74]   93.153909  28.42  100.33
[75+]     92.756186  24.00  101.49

#### Seeing that for the lowest age group [0-14] on average 24.84 of the people were merried is surprising. Some other mean or max values are also unexpected in the lower age groups. These all need to be checked

In [17]:
smam.head()

Country or area  YearStart  YearEnd    Sex  DataValue_ave Region Name  \
0     Afghanistan       1972     1974    Men           26.0        Asia   
1     Afghanistan       1979     1979    Men           25.3        Asia   
2     Afghanistan       2015     2016    Men           24.5        Asia   
3     Afghanistan       1972     1974  Women           18.1        Asia   
4     Afghanistan       1979     1979  Women           17.8        Asia   

  Sub-region Name  
0   Southern Asia  
1   Southern Asia  
2   Southern Asia  
3   Southern Asia  
4   Southern Asia

In [18]:
#I want to check the aggregation for Sex, Sub-region Name
smam.groupby('Sex').agg({'DataValue_ave': ['mean', 'min', 'max']})

DataValue_ave            
               mean   min   max
Sex                            
Men       28.199469  20.8  36.9
Women     24.503577  14.4  34.1

#### The min smam (mean age at first marriage) for women is alarming again and on the otherhand I didn't expect to be as high as 34-37, not even the latest years

In [19]:
#I want to check the aggregation for Sex, Sub-region Name
smam.groupby('Sub-region Name').agg({'DataValue_ave': ['mean', 'min', 'max']})

DataValue_ave            
                                         mean   min   max
Sub-region Name                                          
Australia and New Zealand           27.845455  21.5  32.0
Central Asia                        23.109375  20.3  27.6
Eastern Asia                        27.192492  21.6  33.9
Eastern Europe                      27.028596  20.8  35.7
Latin America and the Caribbean     25.416011  15.9  36.5
Melanesia                           24.918519  19.4  34.3
Micronesia                          24.155882  20.2  30.1
Northern Africa                     26.159770  18.0  33.4
Northern America                    28.621474  21.5  34.9
Northern Europe                     28.987398  21.3  36.3
Polynesia                           26.357732  20.7  35.8
South-eastern Asia                  24.501462  19.3  30.4
Southern Asia                       22.453717  16.4  28.1
Southern Europe                     28.317068  20.6  35.9
Sub-Saharan Africa                  23.782196  14.4  36.9
Western Asia                        25.452581  18.0  32.3
Western Europe                      28.294875  21.4  34.3

#### Central Asia is the only Sub-region where the smam max is below 30 and unfortunaltey we do have a lot of sub-regions where the min starts with a 1, or even as low as 14. It is also interesting to compare all values of the different regions to eachother.

In [20]:
cm.head()

Country or area  YearStart  YearEnd  Sex AgeGroup  DataValue  \
0     Afghanistan       1972     1974  Men  [15-19]        7.6   
1     Afghanistan       1972     1974  Men  [20-24]       31.9   
2     Afghanistan       1972     1974  Men  [25-29]       59.1   
3     Afghanistan       1972     1974  Men  [30-34]       78.9   
4     Afghanistan       1972     1974  Men  [35-39]       87.2   

           Data Source Region Name Sub-region Name  
0  National statistics        Asia   Southern Asia  
1  National statistics        Asia   Southern Asia  
2  National statistics        Asia   Southern Asia  
3  National statistics        Asia   Southern Asia  
4  National statistics        Asia   Southern Asia

In [22]:
#for the currently married ds I want to check the aggregation for, Age-Group Sex, Sub-region Name
cm.groupby('AgeGroup').agg({'DataValue': ['mean', 'min', 'max']})

DataValue               
               mean    min     max
AgeGroup                          
[0-14]    23.061667   0.01   82.01
[10-14]    0.411528   0.00   83.17
[12-14]    3.398000   2.10    5.14
[15-19]    7.119957   0.00   94.50
[16-19]    0.250000   0.16    0.34
[20-24]   30.255166   0.00   96.20
[25-29]   56.336537   3.21   98.60
[30-34]   71.598101   2.17   98.44
[35-39]   77.522102  28.57  100.00
[40-44]   78.998808  36.65  100.00
[45+]     73.980000  57.83   90.13
[45-49]   78.694685  36.61   99.91
[50+]     70.043571  40.80   94.10
[50-54]   77.172627  31.31  100.00
[55+]     65.676667  33.90   95.21
[55-59]   75.188585  27.07  100.00
[60+]     61.477812  26.40   89.52
[60-64]   70.618072  21.13  100.00
[65+]     55.015896  11.20   94.60
[65-69]   66.167525  17.72  100.00
[70+]     50.646803   8.80   92.82
[70-74]   59.092207   8.00   98.94
[75+]     44.670614   6.80   93.70

In [23]:
#for the currently married ds I want to check the aggregation for, Age-Group Sex, Sub-region Name
cm.groupby('Sex').agg({'DataValue': ['mean', 'min', 'max']})

DataValue             
            mean  min     max
Sex                          
Men    61.229148  0.0  100.00
Women  55.224193  0.0   99.91

#### Why does the value here higher for Men than Women? Could this be men dying sooner?

In [25]:
#for the currently married ds I want to check the aggregation for, Age-Group Sex, Sub-region Name
cm.groupby('Sub-region Name').agg({'DataValue': ['mean', 'min', 'max']})

DataValue              
                                      mean   min     max
Sub-region Name                                         
Australia and New Zealand        55.841387  0.00   88.84
Central Asia                     68.607517  0.00   98.53
Eastern Asia                     62.540337  0.00   99.04
Eastern Europe                   55.605283  0.00   95.14
Latin America and the Caribbean  54.666394  0.00   93.30
Melanesia                        60.781929  0.00   96.40
Micronesia                       63.261408  0.02   96.67
Northern Africa                  64.705561  0.00   99.10
Northern America                 49.558423  0.00   88.65
Northern Europe                  49.888446  0.00   92.41
Polynesia                        60.201542  0.00  100.00
South-eastern Asia               64.035624  0.00   98.40
Southern Asia                    69.287046  0.00   99.40
Southern Europe                  57.356869  0.00   98.70
Sub-Saharan Africa               63.368465  0.00  100.00
Western Asia                     65.145775  0.00   99.50
Western Europe                   54.211616  0.00   91.02

#### Not that suprisingly Northern America and Northern Europe has the lowest avg currenly married values, but Southern Asia has the highest, which would be interesting to look into. 